llvm/include/llvm/CodeGen/SDPatternMatch.h

```diff
@@ -695,6 +726,13 @@ template <typename Opnd> inline UnaryOpc_match<Opnd> m_ZExt(const Opnd &Op) {
   return UnaryOpc_match<Opnd>(ISD::ZERO_EXTEND, Op);
 }
 
+template <typename Opnd>
+inline UnaryOpc_match<Opnd> m_NNegZExt(const Opnd &Op) {
+  SDNodeFlags Flags{};
```

Comment:
nit: IIUC since SDNodeFlags has a user-provided default ctor, the value initialization here will eventually fallback to default initialization, which means it's the same as `SDNodeFlags Flags;`. Maybe we can use the latter instead.

ditto for other occurrences in this PR.